### PR TITLE
Updated dependencies and allow PHP 8.1

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Welp, an installation with `config.platform` is kinda needed 😬

**Comment by kukulich** [comment]
Oh, shit, I’ve made it again…

**Comment by Ocramius** [comment]
Happens every time to me as well - it's the problem with supporting multiple releases, heh.

Also here, strictly related to the fact that I've not re-enabled Dependabot since it went EOL at the end of August. Upgrades become more manageable with that, and don't really require human intervention other than dropping support for older PHP versions.

**Comment by kukulich** [comment]
Hmm, it's probably the new GitHub feature... My PRs workflows need to be approved before first PR is merged...

**Comment by Ocramius** [comment]
Yes, until the first contribution is merged, approval is required.

That's because of cryptocurrency miners and their vicious cycle of evilness (people abusing CI workers)

File: .phpcs-cache [hunk 1]
```diff
@@ -0,0 +1 @@
...y":0,"fixable":false}]}},"warnings":[],"metrics":{"Declarations and side effects mixed":{"values":{"no":1}},"PHP short open tag used":{"values":{"no":1}},"EOL char":{"values":{"\\n":1}},"Number of newlines at EOF":{"values":{"1":1}},"PHP closing tag at end of PHP-only file":{"values":{"no":1}},"Line length":{"values":{"80 or less":16}},"Line indent":{"values":{"spaces":7}},"PHP keyword case":{"values":{"lower":13}},"Multiple statements on same line":{"values":{"no":8}},"One class per file":{"values":{"yes":1}},"Class defined in namespace":{"values":{"yes":1}},"PascalCase class name":{"values":{"yes":1}},"Class opening brace placement":{"values":{"new line":1}},"CamelCase method name":{"values":{"yes":1}},"PHP type case":{"values":{"lower":1}},"Function opening brace placement":{"values":{"new line":1}},"Function has doc comment":{"values":{"no":1}},"Function spacing after last":{"values":[1]},"Function spacing before first":{"values":[1]},"Spacing before object operator":{"values":[4]},"Spacing after object operator":{"values":[4]},"Short array syntax used":{"values":{"yes":1}},"Array end comma":{"values":{"no":1}}},"errorCount":1,"warningCount":0,"fixableCount":0,"numTokens":150}}
```
Comment (Ocramius) [inline .phpcs-cache]: Cache ended up being committed?

File: .phpunit.result.cache [hunk 2]
```diff
@@ -0,0 +1 @@
...cAnalysisAgainstMutantTest::testWillConsiderMutantValidIfNoErrorsAreDetectedByStaticAnalysis":0.51500000000000001,"Roave\\InfectionStaticAnalysisTest\\Psalm\\RunStaticAnalysisAgainstMutantTest::testWillConsiderMutantReferencingReflectionApiAsValid":0.27200000000000002,"Roave\\InfectionStaticAnalysisTest\\Psalm\\RunStaticAnalysisAgainstMutantTest::testWillConsiderMutantInvalidIfErrorsAreDetectedByStaticAnalysis":0.23799999999999999,"Roave\\InfectionStaticAnalysisTest\\Psalm\\RunStaticAnalysisAgainstMutantTest::testWillConsiderMutantReferencingProjectFilesAsValid":0.27300000000000002,"Roave\\InfectionStaticAnalysisTest\\Stub\\ArrayFilterTest::testListContentsArePreserved":0.001,"Roave\\InfectionStaticAnalysisTest\\BootstrapperTest::testWillNotTestAnything":1.52,"Roave\\InfectionStaticAnalysisTest\\RunStaticAnalysisAgainstEscapedMutantTest::testWillSkipValidationOnAlreadyKilledMutants":0.0050000000000000001,"Roave\\InfectionStaticAnalysisTest\\RunStaticAnalysisAgainstEscapedMutantTest::testWillNotKillMutantsThatEscapedAndPassedStaticAnalysis":0,"Roave\\InfectionStaticAnalysisTest\\RunStaticAnalysisAgainstEscapedMutantTest::testWillKillMutantsThatEscapedAndFailedStaticAnalysis":0.001}}
```
Comment (Ocramius) [inline .phpunit.result.cache]: Also here: ended up in commit 🤔

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .phpunit.result.cache (hunk 2 above)
Comment (kukulich) [reply .phpunit.result.cache]: Ok, third try... Force pushed.

**Review by Ocramius — APPROVED** [review]
LGTM 👍

**Pull request merged**